Formula/papilo.rb

```diff
@@ -0,0 +1,36 @@
+class Papilo < Formula
+  desc "Parallel Presolve for Integer and Linear Optimization"
+  homepage "https://github.com/scipopt/papilo"
+  url "https://github.com/scipopt/papilo/archive/refs/tags/v2.1.2.tar.gz"
+  sha256 "7e3d829c957767028db50b5c5085601449b00671e7efc2d5eb0701a6903d102f"
+  license all_of: ["LGPL-3.0-only", "GPL-3.0-only"]
+
+  depends_on "cmake" => :build
+  depends_on "gcc" => :build
+  depends_on "boost"
+  depends_on "gmp"
+  depends_on "openblas"
+  depends_on "tbb"
+
+  def install
+    cmake_args = %w[
+      -DBOOST=ON
+      -DGMP=ON
+      -DLUSOL=ON
+      -DQUADMATH=ON
+      -DTBB=ON
+      -DBLA_VENDOR=OpenBLAS
+    ]
+
+    system "cmake", "-B", "papilo-build", "-S", ".", *cmake_args, *std_cmake_args
+    system "cmake", "--build", "papilo-build"
+    system "cmake", "--install", "papilo-build"
+
+    prefix.install "papilo-build/test"
+  end
+
+  test do
+    output = shell_output("#{prefix}/test/unit_test")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
unit_test is a binary that has 102 test cases with overall 1200 assertions. It also includes an input file (here .mps file). Isn‘t this enough?

Reply:
I have updated the test. It does run successfully. But I get this:
```
papilo:
[30](https://github.com/Homebrew/homebrew-core/actions/runs/3926252502/jobs/6711841125#step:6:31)
    * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
[31](https://github.com/Homebrew/homebrew-core/actions/runs/3926252502/jobs/6711841125#step:6:32)
  Error: 1 problem in 1 formula detected
```
I will use papilo as a library for [scip](https://github.com/scipopt/scip) that is notable.